packages/amplify_api_plugin_interface/lib/amplify_api_graphql_helpers_interface.dart

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+// TODO: Datastore dependencies temporarily added in API. Eventually they should be moved to core or otherwise reconciled to avoid duplication.
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+
+import 'src/types.dart';
+import 'src/GraphQL/GraphQLRequest.dart';
+import 'src/types/pagination/model_pagination.dart';
+import 'src/types/pagination/paginated_result.dart';
+
+abstract class ModelQueriesInterface {
+  // Get
+  GraphQLRequest<T> get<T extends Model>(ModelType<T> modelType, String id) {
+    throw UnimplementedError("get() has not been implemented.");
```

Comment:
For abstract classes, we don't want to provide a default impl which throws because subclasses which extend this will inherit that default behavior without warning. Leaving it as just the function signature, like below, means that subclasses will be forced to provide an implementation for all methods.

```dart
GraphQLRequest<T> get<T extends Model>(ModelType<T> modelType, String id);
```

Reply:
That makes sense. We're doing the same thing in the [plugin interface](https://github.com/aws-amplify/amplify-flutter/blob/c0ccdfdb535bf9679d442c9cb9ddccaf9149df4d/packages/amplify_api_plugin_interface/lib/amplify_api_plugin_interface.dart#L29). Does consistency between the two matter?

Reply:
Discussed this offline, we'll be removing the exception from these interfaces to avoid potential confusion when implementing subclasses. The plugin interface needs the extra context as a plugin could be used on an unsupported platform and the exception provides missing context.